custom_components/chargeamps/sensor.py

```diff
@@ -81,35 +123,57 @@ async def async_update(self):
         if not self._interviewed:
             await self.interview()
 
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self._state
+class ChargeampsPowerSensor(ChargeampsSensor):
+    """Chargeamps Power Sensor class."""
 
-    @property
-    def device_state_attributes(self):
-        """Return the state attributes of the sensor."""
-        return self._attributes
+    async def async_update(self):
+        """Update the sensor."""
+        _LOGGER.debug(
+            "Update chargepoint %s connector %s",
+            self.charge_point_id,
+            self.connector_id,
+        )
+        await self.handler.update_data(self.charge_point_id)
+        _LOGGER.debug(
+            "Finished update chargepoint %s connector %s",
+            self.charge_point_id,
+            self.connector_id,
+        )
+        measurements = self.handler.get_connector_measurements(
+            self.charge_point_id, self.connector_id
+        )
+        if measurements:
+            self._state = round(
+                sum([phase.current * phase.voltage for phase in measurements]), 0
+            )
+            self._attributes["Active_Phase"] = " ".join(
+                [i.phase for i in measurements if i.current > 0]
+            )
+            for measure in measurements:
+                self._attributes[f"{measure.phase}_power_(W)"] = round(
+                    measure.voltage * measure.current, 0
+                )
+                self._attributes[f"{measure.phase}_current_(A)"] = round(
+                    measure.current, 1
+                )
+        else:
+            self._attributes["Active_Phase"] = ""

```

Comment:
"" or None?

Reply:
I'm thinking that templates expect a string here.

Reply:
```suggestion
            self._attributes["active_phase"] = ""

```